### PR TITLE
Ingest EMNLP19

### PR DESCRIPTION
**Pull request by mjpost**

### CODE REVIEW

**Comment by mjpost** [comment]
Does anyone know how attachments are labeled in ACLPUB? EMNLP's are all named like this:

    <attachment type="optional-supplementary-material">D19-1001.OptionalSupplementaryMaterial.pdf</attachment>

which loses the typing information. Is this a setting in ACLPUB or part of Softconf?

CC: @davidweichiang @mbollmann @rrgerber

**Comment by mjpost** [comment]
(And specifically: is this something that it is too late to fix?)

File: bin/anthology/papers.py [hunk 1]
```diff
@@ -156,19 +157,11 @@ def volume_id(self):
 
     @property
     def paper_id(self):
-        if self.collection_id[0] == "W" or self.collection_id == "C69":
```
Comment (akoehn) [inline bin/anthology/papers.py]: This changes the semantics of paper_id if I understand correctly. Could that change anything in non-obvious places?

Maybe this would be a good point to make naming consistent: `paper_ids` in volume return full ids and should therefore probably be renamed to `full_paper_ids`.

This is not blocking the merge, just want to note it before it is forgotten again.

File: bin/anthology/utils.py [hunk 2]
```diff
@@ -37,6 +39,7 @@ def is_volume_id(anthology_id):
         anthology_id[-3:] == "000"
         or (anthology_id[0] == "W" and anthology_id[-2:] == "00")
         or (anthology_id[:3] == "C69" and anthology_id[-2:] == "00")
+        or (anthology_id[:3] == 'D19' and anthology_id[-2:] == '00')
```
Comment (mbollmann) [inline bin/anthology/utils.py]: Wouldn't that make e.g. `D19-1100` a volume ID, which it is not?

File: bin/anthology/papers.py (hunk 1 above)
Comment (mjpost) [reply bin/anthology/papers.py]: I looked at all calls to paper ID and updated them.

IMO, `paper_id` should return the paper number, analogous to `anthology_id` and `collection_id`. Then, `full_id` should be renamed `anthology_id`.

Comment (mjpost) [reply bin/anthology/papers.py]: I think this is resolved.

**Comment by mjpost** [comment]
This should be ready. The tutorial abstracts are going to be submitted later.

File: bin/anthology/utils.py [hunk 3]
```diff
@@ -14,15 +14,18 @@
...
 
-from .people import PersonName
-from . import data
+from lxml import etree
+from urllib.parse import urlparse
+from xml.sax.saxutils import escape as xml_escape
+from typing import Tuple, Optional
+
+from anthology.people import PersonName
+from anthology import data
```
Comment (mbollmann) [inline bin/anthology/utils.py]: What's the reason for this change? It doesn't seem right to me?

File: bin/anthology/volumes.py [hunk 4]
```diff
@@ -21,7 +21,7 @@
 from .papers import Paper
 from .venues import VenueIndex
 from .sigs import SIGIndex
-from anthology.utils import parse_element, is_journal, month_str2num, infer_url, infer_year
+from anthology.utils import build_anthology_id, parse_element, is_journal, month_str2num, infer_url, infer_year
```
Comment (mbollmann) [inline bin/anthology/volumes.py]: Same here, why isn't this relative? (`from .utils`?)

File: bin/anthology/utils.py (hunk 3 above)
Comment (mjpost) [reply bin/anthology/utils.py]: I had the idea that there was a move from dot-style relative imports to using absolute imports, even within a package, based on an import problem I had a while ago that was resolved by making this change. But I can't find any documentation for this so I'll revert it.
